src/gpio.rs

```diff
@@ -310,17 +310,17 @@ macro_rules! gpio {
                     fn mode<M: PinMode>(&mut self) {
                         let offset = 2 * $i;
                         unsafe {
-                            &(*$GPIOX::ptr()).pupdr.modify(|r, w| {
+                            let _ = &(*$GPIOX::ptr()).pupdr.modify(|r, w| {
```

Comment:
Hmm, can't the `&` just be removed, instead of using the borrow? Same for the other places where the warning occurred.

Just pondering. Feel free to look into it, if you want to, but this is good enough. It can be cleaned up some other time.

Reply:
Good thinking, I like that way better. Will update!